tests/visualization_tests/test_optimization_history.py

```diff
@@ -89,63 +91,79 @@ def objective(trial: Trial) -> float:
+            _ValuesInfo(best_values, None, "Best Value"),
+        )
+    ]
 
     # Test customized target.
-    with pytest.warns(UserWarning):
-        figure = plot_optimization_history(studies, target=lambda t: t.number)
-    assert len(figure.data) == 1 * n_studies
-    assert np.array_equal(figure.data[0].x, [0, 1, 2])
-    assert np.array_equal(figure.data[0].y, [0, 1, 2])
-
-    # Test customized target name.
-    custom_target_name = "Target Name"
-    figure = plot_optimization_history(studies, target_name=custom_target_name)
-    expected_legend_texts = []
-    for i in range(n_studies):
-        expected_legend_texts.append(f"Best Value of {studies[i].study_name}")
-        expected_legend_texts.append(f"{custom_target_name} of {studies[i].study_name}")
-    legend_texts = [scatter.name for scatter in figure.data]
-    assert sorted(legend_texts) == sorted(expected_legend_texts)
-    assert figure.layout.yaxis.title.text == custom_target_name
-
-    # Ignore failed trials.
-    studies = [create_study(direction=direction) for _ in range(n_studies)]
-    for study in studies:
-        study.optimize(fail_objective, n_trials=1, catch=(ValueError,))
-
-    figure = plot_optimization_history(studies)
-    assert len(figure.data) == 0
+    info_list = _get_optimization_history_info_list(
+        study, target=lambda t: t.number, target_name=target_name, error_bar=False
+    )
+    assert info_list == [
+        _OptimizationHistoryInfo(
+            [0, 1, 2],
+            _ValuesInfo([0.0, 1.0, 2.0], None, target_name),
+            None,
+        )
+    ]
 
 
 @pytest.mark.parametrize("direction", ["minimize", "maximize"])
-def test_plot_optimization_history_with_error_bar(direction: str) -> None:
+@pytest.mark.parametrize("target_name", ["Objective Value", "Target Name"])
+def test_get_optimization_history_info_list_with_multiple_studies(
+    direction: str, target_name: str
+) -> None:
     n_studies = 10
+    base_values = [1.0, 2.0, 0.0]
+    base_best_values = [1.0, 1.0, 0.0] if direction == "minimize" else [1.0, 2.0, 2.0]
 
-    # Test with no trial.
+    # Test with trials.
     studies = [create_study(direction=direction) for _ in range(n_studies)]
-    figure = plot_optimization_history(studies, error_bar=True)
-    assert len(figure.data) == 0
+    for i, study in enumerate(studies):
+        study.optimize(lambda t: base_values[t.number] + i, n_trials=3)
+    info_list = _get_optimization_history_info_list(
+        studies, target=None, target_name=target_name, error_bar=False
+    )
+
+    for i, info in enumerate(info_list):
+        values_i = [v + i for v in base_values]
+        best_values_i = [v + i for v in base_best_values]
+        assert info == _OptimizationHistoryInfo(
+            [0, 1, 2],
+            _ValuesInfo(values_i, None, f"{target_name} of {studies[i].study_name}"),
+            _ValuesInfo(best_values_i, None, f"Best Value of {studies[i].study_name}"),
+        )
```

Comment:
Notes: Previously, all studies had the same optimization history. The current code changes introduce the offset (`i`) to differentiate optimization history of the studies.